### ISSUE TITLE
README.md can be improved

### ISSUE THREAD

**Issue by schniepp** [strukturag/pdfdraw#3]
This sounds like a really interesting app! This addresses an important need. So thanks for working on this!

I have a lot of questions, and I did not find enough information about it in the README.md file (or anywhere else). 

Questions are:
1. Where is the annotation information stored? In the NC database? In a separate file? In the PDF itself?
1. Related to that, what permissions are required? Does the PDF file need to be writable? Does the folder need write permission? (I am asking because I have a significant number of PDFs on an externally mounted folder, which does not give users write permit.)
1. Also related to the first question: what happens if the PDF file gets moved or renamed (or deleted, for that matter)?
1. Where can I find more details about the required node.js installation?

Thanks for your support?

**Comment by fancycode** [comment]
Thanks, here are some answers.

> 1. Where is the annotation information stored? In the NC database? In a separate file? In the PDF itself?

The annotations are stored in the NC database but you can generate an annotated PDF.

> 2. Related to that, what permissions are required? Does the PDF file need to be writable? Does the folder need write permission? (I am asking because I have a significant number of PDFs on an externally mounted folder, which does not give users write permit.)

Adding annotations requires write permissions right now (as you are "writing" to the document). Users with read permissions can view a document being annotated. I'm open for suggestions / PRs that change that behaviour.

> 3. Also related to the first question: what happens if the PDF file gets moved or renamed (or deleted, for that matter)?

The annotations reference the PDF by its database id, so renaming / moving works. When a PDF is deleted, the annotations are deleted, too.

> 4. Where can I find more details about the required node.js installation?

The node.js server is located in this repository in the `server` folder: https://github.com/strukturag/pdfdraw/tree/master/server

**Comment by schniepp** [comment]
Thanks so much, that sounds all pretty good!

I think I understand all answers, except for one: if the annotations are written to the db, then why is write permission to the document needed?

I haven't installed the node.js server yet, but I want to test it, soon. Especially when the generated annotated PDF can be stored in a folder of choice, this would be extremely helpful. I am thinking this could go around the problem where there are not write permissions to the original PDF or its containing folder.

**Comment by fancycode** [comment]
That's not filesystem write permissions but Nextcloud permissions, e.g. if you share a folder with somebody but don't allow edits in the share, people you share with can't annotate PDFs.

The generated annotated PDF is downloaded, so you can save it anywhere.

**Comment by TheNomad11** [comment]
I agree that the Readme needs to be improved. Non-sysadmin-friendly installation instructions of the node.js server would be helpful as well. According to the github page of the nodejs.server it seems to be a complex process to install it. This app could be so helpful and valuable for many users! Thanks for developping it!!

**Comment by schniepp** [comment]
@fancycode : OK, thanks! So that means that when a user does not have write permits to a file/folder any edits to the DB are also not allowed when there are not edit permissions to the file system? I did not know that. So that means users also cannot comment, etc. on a file unless they have edit permits to that folder?
I guess my confusion comes because there are two levels of permissions: one at the file system level, and one at the NC level. For instance, I could share an external file system in NC to which even the NC server only has read permits. The permission set within NC might be totally separate from that. I guess you were talking about the permission settings within NC, whereas I also thought about permissions on the file system level.

**Comment by s1787956** [comment]
Hi very nice app! I kind of got it working but I could not follow the nginx step of the README.md. Therefore I cannot download the annotated pdf. I guess I am using Apache not nginx. Any idea what I should do?

Anyway Thanks for the great app!

**Comment by julienym** [comment]
Yes very nice, and seams like working well in NC 21. One question, I can't seam to select text for highlighting, is that normal ?
